mico-core/src/main/java/io/github/ust/mico/core/broker/MicoServiceInterfaceBroker.java

```diff
@@ -1,4 +1,96 @@
+    }
+
+    public List<String> getPublicIpsOfInterfaceByInterfaceName(String shortName, String version, String serviceInterfaceName, io.fabric8.kubernetes.api.model.Service kubernetesService) {
+        List<String> publicIps = new ArrayList<>();
+        LoadBalancerStatus loadBalancer = kubernetesService.getStatus().getLoadBalancer();
+
+        if (loadBalancer != null) {
+            List<LoadBalancerIngress> ingressList = loadBalancer.getIngress();
+            if (ingressList != null && !ingressList.isEmpty()) {
+                log.debug("There is/are {} ingress(es) defined for Kubernetes service '{}' (MicoServiceInterface '{}').",
+                        ingressList.size(), kubernetesService.getMetadata().getName(), serviceInterfaceName);
+                for (LoadBalancerIngress ingress : ingressList) {
+                    publicIps.add(ingress.getIp());
+                }
+                log.info("Service interface with name '{}' of MicoService '{}' in version '{}' has external IPs: {}",
+                        serviceInterfaceName, shortName, version, publicIps);
+            }
+        }
+
+        return publicIps;
+    }
+
+    public void deleteMicoServiceInterface(String shortName, String version, String serviceInterfaceName) {
+        serviceInterfaceRepository.deleteByServiceAndName(shortName, version, serviceInterfaceName);
+    }
+
+    public MicoServiceInterface persistMicoServiceInterface(MicoService micoService, MicoServiceInterface micoServiceInterface) throws MicoServiceInterfaceAlreadyExistsException, MicoServiceNotFoundException {
+        Optional<MicoServiceInterface> micoServiceInterfaceOptional = serviceInterfaceRepository.findByServiceAndName(micoService.getShortName(), micoService.getVersion(), micoServiceInterface.getServiceInterfaceName());
+
+        if (micoServiceInterfaceOptional.isPresent()) {
+            throw new MicoServiceInterfaceAlreadyExistsException(micoService.getShortName(), micoService.getVersion(), micoServiceInterface.getServiceInterfaceName());
+        }
+
+        micoService.getServiceInterfaces().add(micoServiceInterface);
+        micoServiceBroker.updateExistingService(micoService);
+
+        return micoServiceInterface;
+    }
+
+    public MicoServiceInterface updateMicoServiceInterface(String shortName, String version, String serviceInterfaceName, MicoServiceInterface micoServiceInterface) throws MicoServiceInterfaceNotFoundException {
+        Optional<MicoServiceInterface> serviceInterfaceOptional = serviceInterfaceRepository.findByServiceAndName(shortName, version, serviceInterfaceName);
+
+        if (!serviceInterfaceOptional.isPresent()) {
+            throw new MicoServiceInterfaceNotFoundException(shortName, version, serviceInterfaceName);
+        }
+
+        MicoServiceInterface serviceInterface = serviceInterfaceOptional.get();
+        MicoServiceInterface updatedServiceInterface = micoServiceInterface.setId(serviceInterface.getId());
```

Comment:
Question:
Is it required to also use the ID of the already existing ports if an interface is updated?
→ Nodes of the existing ports in the database will become orphans (without edges)?
 @JakobBenz